frontend/packages/core-api-client/src/lib/client/Record.ts

```diff
@@ -16,19 +13,29 @@ export class RecordClient {
     this.restClient = restClient;
   }
 
-  create = (request: RecordCreateRequest): Promise<RecordCreateResponse> => {
-    return this.restClient.do(request, '/records', 'post', request.object, 200);
+  create = async (record: RecordDefinition): Promise<Record> => {
+    const apiResponse = await this.restClient.post('/records', record);
+    if (apiResponse.success) {
+      return apiResponse.response as Record
+    }
+    return apiResponse.error
```

Comment:
Returning `apiResponse.error` should give the incorrect type as it's expecting `Promise<Record>`. `apiResponse.error` has the type `any` so typescript allows this.

It should either return `Promise<Record | ErrorThing>` or throw an error.

I would suggest leaving this file as it was and handling the success check in the recipient client.